cmake/debugflags.cmake

```diff
@@ -15,6 +15,14 @@ set(FLAGS
     # Disable unused parameter warnings (most cases are on purpose):
     -Wno-unused-parameter
 
+    # range checks for C++ datatypes:
+    -D_GLIBCXX_ASSERTIONS
+
+    # more stack protection:
+    -fstack-protector-strong
+    -fcf-protection
+    -fstack-clash-protection
```

Comment:
I think that flag is only available with gcc, that is why I put it further down in this file #1062
(But Travis seems to be asleep right now)

Reply:
Ah, I see, thanks! I've moved the flags to the GNU-section

Reply:
-fstack-protector-strong is available in both gcc and clang, only fstack-clash-protection is gcc only.
Not sure about cf-protection.

Sorry, should've been more specific

Reply:
I think it's fine if we stay safe here (not sure whether there are any c++ compilers besider clang and gcc that possibly don't support it..).

Reply:
Alright!